modules/core/src/main/scala/org/tessellation/domain/snapshot/programs/Download.scala

```diff
@@ -253,7 +258,8 @@ object Download {
         }
 
     def fetchSnapshot(hash: Option[Hash], ordinal: SnapshotOrdinal): F[Signed[GlobalIncrementalSnapshot]] =
-      clusterStorage.getPeers
+      clusterStorage.getResponsivePeers
+        .map(NodeState.ready(_))
```

Comment:
I think I saw getting responsive peers and filtering only ready ones in couple places in the code. Maybe it's time to add a separate method for that? `getResponsiveReadyPeers`?

Reply:
You can have `NodeState.ready` here without (_)